src/pages/Fox/components/FoxWifHatClaim/FoxWifHatClaimConfirm.tsx

```diff
@@ -0,0 +1,256 @@
+import {
+  Button,
+  Card,
+  CardBody,
+  CardFooter,
+  CardHeader,
+  Flex,
+  Skeleton,
+  Stack,
+} from '@chakra-ui/react'
+import type { AccountId } from '@shapeshiftoss/caip'
+import { foxWifHatAssetId, fromAccountId } from '@shapeshiftoss/caip'
+import { CONTRACT_INTERACTION } from '@shapeshiftoss/chain-adapters'
+import { FOX_WIF_HAT_MERKLE_DISTRIBUTOR_ABI } from '@shapeshiftoss/contracts'
+import { useMutation } from '@tanstack/react-query'
+import type { FC } from 'react'
+import { useCallback, useMemo } from 'react'
+import { useTranslate } from 'react-polyglot'
+import { useHistory } from 'react-router'
+import { encodeFunctionData } from 'viem'
+import { Amount } from 'components/Amount/Amount'
+import { AssetIcon } from 'components/AssetIcon'
+import type { RowProps } from 'components/Row/Row'
+import { Row } from 'components/Row/Row'
+import { SlideTransition } from 'components/SlideTransition'
+import { Timeline, TimelineItem } from 'components/Timeline/Timeline'
+import { useEvmFees } from 'hooks/queries/useEvmFees'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import { fromBaseUnit } from 'lib/math'
+import { firstFourLastFour } from 'lib/utils'
+import {
+  assertGetEvmChainAdapter,
+  buildAndBroadcast,
+  createBuildCustomTxInput,
+} from 'lib/utils/evm'
+import { FOX_WIF_HAT_MERKLE_DISTRIBUTOR_CONTRACT } from 'pages/Fox/constant'
+import { useFoxWifHatMerkleTreeQuery } from 'pages/Fox/hooks/useFoxWifHatMerkleTreeQuery'
+import {
+  selectAccountNumberByAccountId,
+  selectAssetById,
+  selectTxById,
+} from 'state/slices/selectors'
+import { serializeTxIndex } from 'state/slices/txHistorySlice/utils'
+import { useAppSelector } from 'state/store'
+
+import { FoxWifHatClaimRoutePaths } from './types'
+
+type FoxWifHatClaimConfirmProps = {
+  setClaimTxid: (txId: string) => void
+  claimTxid: string | undefined
+  accountId: AccountId
+}
+
+const CustomRow: React.FC<RowProps> = props => <Row fontSize='sm' fontWeight='medium' {...props} />
+
+export const FoxWifHatClaimConfirm: FC<FoxWifHatClaimConfirmProps> = ({
+  accountId,
+  claimTxid,
+  setClaimTxid,
+}) => {
+  const history = useHistory()
+  const translate = useTranslate()
+  const wallet = useWallet().state.wallet
+  const accountNumber = useAppSelector(state =>
+    selectAccountNumberByAccountId(state, { accountId }),
+  )
+  const foxWifHatAsset = useAppSelector(state => selectAssetById(state, foxWifHatAssetId))
+  const getFoxWifHatClaimsQuery = useFoxWifHatMerkleTreeQuery()
+
+  const claimQuote = useMemo(() => {
+    const claim = getFoxWifHatClaimsQuery.data?.claims[fromAccountId(accountId).account]
+    if (!claim) return null
+
+    return claim
+  }, [getFoxWifHatClaimsQuery.data, accountId])
+
+  const amountCryptoPrecision = useMemo(() => {
+    if (!claimQuote) return
+    if (!foxWifHatAsset) return
```

Comment:
suggestion: also early return from JSX render

Reply:
Not so useful, did it in [7e9a198](https://github.com/shapeshift/web/pull/8611/commits/7e9a198a552c9ffbe24675fdb83a91d1875e342b)